### PR TITLE
Simple fixes

### LINKED ISSUE #893: CI has excessive timeout / does not cancel redundant jobs
**Problem description**

The new GitHub actions CI appears to not time out (or only after an exorbitantly long time - I've seen runs beyond 16 hours at this point), and there may not be a way for contributors to stop runners once they are deployed.

**Steps to reproduce**

Presumably, any unit test with an infinite loop will trigger this behavior.

**Behavior**

Runners become unresponsive; an attempt to terminate by pushing a new/corrected version will result in the new version remaining queued.

**System information**

N/A

**Additional context**

How to [specify a timeout for GitHub actions](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes)

Action to [cancel current runners](https://github.com/marketplace/actions/cancel-workflow-action)

Current [Actions overview](https://github.com/Cantera/cantera/actions) on Cantera/cantera

There appear to be several unresponsive runners in #892 that I am unable to terminate.

### LINKED ISSUE #1188: Any compiler failure during configuration shows the message about `std::isnan`
**Problem description**

I had a misconfigured compiler (Conda clang, macOS, missing sysroot configuration, the specifics here aren't too important) and Cantera's configuration step was failing, with the error message about `std::isnan` not being available, introduced in #1161. However, inspecting the `config.log` showed me other errors, mostly that the linker couldn't find a few standard libraries.

The problem is that the failure was _not_ due to missing `std::isnan`, so I think the error message is too eager.

**Steps to reproduce**

I'm not sure what an easy way to reproduce this is... perhaps move one of the standard libraries out of the search path temporarily?

**System information**

- Cantera version: I'm building from a branch off main, but any commit after #1161 was merged should do.
- OS: macOS, but I don't think it's important

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Some simple fixes that address paper-cuts during compilation and CI runs.

For a successfully cancelled CI job (triggered by pushing another job :tada:), see [Cantera/cantera/actions/runs/1862740301](https://github.com/Cantera/cantera/actions/runs/1862740301)

**If applicable, fill in the issue number this pull request is fixing**

Closes #1188, closes #893

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -14,10 +14,15 @@ on:
...
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   ubuntu-multiple-pythons:
     name: ${{ matrix.os }} with Python ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```
Comment (bryanwweber) [inline .github/workflows/main.yml:25]: I see this was added to all the jobs, but the commit message didn't say why 😄 what was your thought here?

**Review by bryanwweber — COMMENTED** [review]
Thanks @ischoegl!

File: .github/workflows/main.yml (hunk 1 above)
Comment (ischoegl) [reply .github/workflows/main.yml:25]: Some time ago there was an instance when a test didn’t converge (before I had access to the ‘cancel job’ button), and it was necessary to wait until GH timed out. For OSX, there are only a limited number of runners, so this created a bottle neck that took literally a day to clear out. 60 minutes is imho plenty of time, as even the slowest runners finish in less than half an hour (exception is Coverage, where I put 90 minutes).

Comment (ischoegl) [reply .github/workflows/main.yml:25]: Added this to the commit comment.

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**